### ISSUE TITLE
Inspection errors should (optionally) cause build to fail

### ISSUE THREAD

**Issue by lassemaatta** [thheller/shadow-cljs#355]
At the moment `shadow-cljs` logs inspection errors but there is no way to force the build(s) to fail when such an error occurs. Naturally some inspection errors may cause a specific build target compilation to fail, but this is not guaranteed. 

For example, a CI test build may have a typo in a specific test namespace. `shadow-cljs` will log an inspection error but the build itself will succeed as the offending namespace is ignored. It would be great if we could specify in `shadow-cljs.edn` (preferably per build) whether inspection errors should immediately abort the build.

**Comment by thheller** [comment]
They do typically fail the build.

Tests builds are the exception since they include matching namespaces by regexp. Didn't think about the problem that this may cause namespaces to not be included. Will see about checking that in tests builds and failing them early.

**Comment by thheller** [comment]
Hmm do you have a reproducible example? I added a "bad" test ns and it failed the test build as it should have?

**Comment by lassemaatta** [comment]
I'll try to extract a minimal example which I can publish. 

In the meantime, the actual case where we encountered this issue was a Clojure test-ns which was accidentally written in a `.cljc`-file and the inspection failed due to `:require [clojure.test :refer :all]`. The configuration was:

```
{:lein   true
 :builds {:ci-test {:target        :karma
                    :output-to     "./ci.js"
                    :build-options {:cache-level :off}}}}
```

**Comment by thheller** [comment]
Seems to fail properly too.

```
------ ERROR -------------------------------------------------------------------
 File: C:\Users\thheller\code\shadow-cljs\src\dev\demo\bad_test.cljc:1:1
--------------------------------------------------------------------------------

   1 | (ns demo.bad-test
-------^------------------------------------------------------------------------
Invalid namespace declaration
-- Syntax error -------------------

  (... ... (:require [clojure.test :refer :all]))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

has extra input

-- Relevant specs -------
...
```

Curious side note: Why did you disable cache?

**Comment by lassemaatta** [comment]
I made a simple example repo which reproduces the issue. However, during testing I noticed that we are using and older version of `shadow-cljs` (`2.3.16`) and the problem no longer occurs in the latest version (`2.4.28`) where an incorrect `ns` declaration now throws a compilation error. Therefore, I think this issue can be closed. Thanks for your help and for this awesome tool!

Side note: Disabling the cache in this case is most likely unnecessary, as the CI build we run are executed on a clean Jenkins slave without retaining any caches between builds.
